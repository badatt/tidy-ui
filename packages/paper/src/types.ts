import { HTMLAttributes } from 'react';

/**
 * Paper props
 *
 * @typedef {IPaperProps}
 * @interface IPaperProps
 * @augments {HTMLAttributes<HTMLDivElement>}
 * @public
 */
export interface IPaperProps extends HTMLAttributes<HTMLDivElement> {
  /**
   * If `true`, the component is created with sharp edges
   *
   * @type {?boolean}
   * @default false
   */
  sharp?: boolean;
}
