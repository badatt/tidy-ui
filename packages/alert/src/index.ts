export * from './Alert';
export * from './AlertFooter';
export * from './AlertTitle';
