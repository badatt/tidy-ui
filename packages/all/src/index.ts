export * from '@tidy-ui/commons';
export * from '@tidy-ui/feedback';
export * from '@tidy-ui/form';
export * from '@tidy-ui/layout';
export * from '@tidy-ui/navigation';
export * from '@tidy-ui/presentation';
