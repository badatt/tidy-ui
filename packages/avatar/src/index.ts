export * from './Avatar';
export * from './AvatarBadge';
export * from './AvatarGroup';
