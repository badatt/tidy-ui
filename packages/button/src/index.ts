export * from './Button';
export * from './Anchor';
