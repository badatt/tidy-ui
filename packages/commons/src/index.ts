export * from './actions';
export * from './hooks';
export * from './icons';
export * from './provider';
export * from './reducers';
export * from './theme';
export * from './utils';
