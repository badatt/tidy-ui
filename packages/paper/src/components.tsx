import { css, styled } from '@tidy-ui/commons';
import { IPaperProps } from './types';

/**
 * Paper root component
 *
 * @internal
 */
const PaperRoot = styled.section<IPaperProps>`
  ${({ theme, sharp }) => css`
    display: flex;
    padding: 1rem;
    background-color: ${theme.palette.background.paper};
    border-radius: ${!sharp && theme.layout.radius};
    box-shadow: ${theme.layout.shadow};
    color: ${theme.palette.text.primary};
    margin-bottom: 1.875rem;
  `}
`;

export { PaperRoot };
