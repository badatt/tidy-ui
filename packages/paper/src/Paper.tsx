import React, { forwardRef } from 'react';
import { PaperRoot } from './components';
import { IPaperProps } from './types';

/**
 * Paper component can be used to represent an elevated section in a page,
 * powered by various styles (with theming) and settings
 *
 */
const Paper = forwardRef<HTMLDivElement, IPaperProps>((props, ref) => {
  const { className, children, ...rest } = props;
  return (
    <PaperRoot role="paper" ref={ref} {...rest} className={className}>
      {children}
    </PaperRoot>
  );
});

Paper.defaultProps = {
  isSharp: false,
};

export { Paper };
