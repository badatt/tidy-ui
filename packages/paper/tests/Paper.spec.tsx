import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

import { orchidLight } from '@tidy-ui/theme';

import { Paper } from '../src';

describe('Paper', () => {
  test('Default render', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={orchidLight}>
          <Paper>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quod, veniam ducimus ea reiciendis sed blanditiis
            impedit quisquam velit praesentium, eius dignissimos. Omnis ad doloribus laudantium expedita accusantium
            placeat suscipit dolores. Autem iste voluptas deleniti similique accusamus voluptatem dolorem necessitatibus
            aut natus facere, tenetur sequi culpa beatae praesentium hic aliquam velit molestias illo enim sint
            repellendus earum quasi est repudiandae. Eveniet?
          </Paper>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Rounded paper', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={orchidLight}>
          <Paper rounded>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quod, veniam ducimus ea reiciendis sed blanditiis
            impedit quisquam velit praesentium, eius dignissimos. Omnis ad doloribus laudantium expedita accusantium
            placeat suscipit dolores. Autem iste voluptas deleniti similique accusamus voluptatem dolorem necessitatibus
            aut natus facere, tenetur sequi culpa beatae praesentium hic aliquam velit molestias illo enim sint
            repellendus earum quasi est repudiandae. Eveniet?
          </Paper>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
