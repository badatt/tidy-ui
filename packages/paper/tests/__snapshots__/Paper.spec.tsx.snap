// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper Basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  background-color: hsla(0,100%,100%,1);
  border-radius: 0.5rem;
  box-shadow: rgb(255,255,255) 0px 0px 0px 0px,rgba(15,23,42,0.05) 0px 0px 0px 1px,rgba(0,0,0,0.1) 0px 20px 25px -5px,rgba(0,0,0,0.1) 0px 8px 10px -6px;
  color: hsla(217,33%,17%,1);
  margin-bottom: 1.875rem;
}

<body>
    <div>
      <section
        class="c0"
        role="paper"
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quod, veniam ducimus ea reiciendis sed blanditiis impedit quisquam velit praesentium, eius dignissimos. Omnis ad doloribus laudantium expedita accusantium placeat suscipit dolores. Autem iste voluptas deleniti similique accusamus voluptatem dolorem necessitatibus aut natus facere, tenetur sequi culpa beatae praesentium hic aliquam velit molestias illo enim sint repellendus earum quasi est repudiandae. Eveniet?
      </section>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  background-color: hsla(0,100%,100%,1);
  border-radius: 0.5rem;
  box-shadow: rgb(255,255,255) 0px 0px 0px 0px,rgba(15,23,42,0.05) 0px 0px 0px 1px,rgba(0,0,0,0.1) 0px 20px 25px -5px,rgba(0,0,0,0.1) 0px 8px 10px -6px;
  color: hsla(217,33%,17%,1);
  margin-bottom: 1.875rem;
}

<div>
    <section
      class="c0"
      role="paper"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quod, veniam ducimus ea reiciendis sed blanditiis impedit quisquam velit praesentium, eius dignissimos. Omnis ad doloribus laudantium expedita accusantium placeat suscipit dolores. Autem iste voluptas deleniti similique accusamus voluptatem dolorem necessitatibus aut natus facere, tenetur sequi culpa beatae praesentium hic aliquam velit molestias illo enim sint repellendus earum quasi est repudiandae. Eveniet?
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Paper Dark mode basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  background-color: hsla(217,33%,17%,1);
  border-radius: 0.5rem;
  box-shadow: rgb(255,255,255) 0px 0px 0px 0px,rgba(15,23,42,0.05) 0px 0px 0px 1px,rgba(0,0,0,0.1) 0px 20px 25px -5px,rgba(0,0,0,0.1) 0px 8px 10px -6px;
  color: hsla(210,40%,96%,1);
  margin-bottom: 1.875rem;
}

<body>
    <div>
      <section
        class="c0"
        role="paper"
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quod, veniam ducimus ea reiciendis sed blanditiis impedit quisquam velit praesentium, eius dignissimos. Omnis ad doloribus laudantium expedita accusantium placeat suscipit dolores. Autem iste voluptas deleniti similique accusamus voluptatem dolorem necessitatibus aut natus facere, tenetur sequi culpa beatae praesentium hic aliquam velit molestias illo enim sint repellendus earum quasi est repudiandae. Eveniet?
      </section>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  background-color: hsla(217,33%,17%,1);
  border-radius: 0.5rem;
  box-shadow: rgb(255,255,255) 0px 0px 0px 0px,rgba(15,23,42,0.05) 0px 0px 0px 1px,rgba(0,0,0,0.1) 0px 20px 25px -5px,rgba(0,0,0,0.1) 0px 8px 10px -6px;
  color: hsla(210,40%,96%,1);
  margin-bottom: 1.875rem;
}

<div>
    <section
      class="c0"
      role="paper"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quod, veniam ducimus ea reiciendis sed blanditiis impedit quisquam velit praesentium, eius dignissimos. Omnis ad doloribus laudantium expedita accusantium placeat suscipit dolores. Autem iste voluptas deleniti similique accusamus voluptatem dolorem necessitatibus aut natus facere, tenetur sequi culpa beatae praesentium hic aliquam velit molestias illo enim sint repellendus earum quasi est repudiandae. Eveniet?
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Paper Paper with sharp edges 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  background-color: hsla(0,100%,100%,1);
  box-shadow: rgb(255,255,255) 0px 0px 0px 0px,rgba(15,23,42,0.05) 0px 0px 0px 1px,rgba(0,0,0,0.1) 0px 20px 25px -5px,rgba(0,0,0,0.1) 0px 8px 10px -6px;
  color: hsla(217,33%,17%,1);
  margin-bottom: 1.875rem;
}

<body>
    <div>
      <section
        class="c0"
        role="paper"
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quod, veniam ducimus ea reiciendis sed blanditiis impedit quisquam velit praesentium, eius dignissimos. Omnis ad doloribus laudantium expedita accusantium placeat suscipit dolores. Autem iste voluptas deleniti similique accusamus voluptatem dolorem necessitatibus aut natus facere, tenetur sequi culpa beatae praesentium hic aliquam velit molestias illo enim sint repellendus earum quasi est repudiandae. Eveniet?
      </section>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  background-color: hsla(0,100%,100%,1);
  box-shadow: rgb(255,255,255) 0px 0px 0px 0px,rgba(15,23,42,0.05) 0px 0px 0px 1px,rgba(0,0,0,0.1) 0px 20px 25px -5px,rgba(0,0,0,0.1) 0px 8px 10px -6px;
  color: hsla(217,33%,17%,1);
  margin-bottom: 1.875rem;
}

<div>
    <section
      class="c0"
      role="paper"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quod, veniam ducimus ea reiciendis sed blanditiis impedit quisquam velit praesentium, eius dignissimos. Omnis ad doloribus laudantium expedita accusantium placeat suscipit dolores. Autem iste voluptas deleniti similique accusamus voluptatem dolorem necessitatibus aut natus facere, tenetur sequi culpa beatae praesentium hic aliquam velit molestias illo enim sint repellendus earum quasi est repudiandae. Eveniet?
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
